services/motion/builtin/builtin.go

```diff
@@ -122,18 +127,31 @@ func (ms *builtIn) Reconfigure(
 	ms.movementSensors = movementSensors
 	ms.slamServices = slamServices
 	ms.components = components
+	if ms.state != nil {
+		ms.state.Stop()
+	}
+	ms.state = state.NewState(ms.logger)
 	return nil
 }
 
 type builtIn struct {
 	resource.Named
-	resource.TriviallyCloseable
 	fsService       framesystem.Service
+	closed          atomic.Bool
```

Comment:
closed tracks whether or not the model has been closed so that all methods can return errors if they have been closed, as expected by the RDK resource system.